File: ros/ws02/first_ws/src/topic_pkg/src/publisher.cpp
//
// Created by zcb on 5/3/21.
//

#include <ros/ros.h>
#include <iostream>
#include <std_msgs/String.h>


int main(int argc, char **argv) {

    std::string nodeName = "publisher_node";
    ros::init(argc, argv, nodeName);

    ros::NodeHandle node;

    std::string topicName = "/zcb01/topic";
    ros::Publisher publisher = node.advertise<std_msgs::String>(topicName, 1000);
    std_msgs::String msg;


    ros::Rate rate(2); // hz = 2
    int idx = 0;
    while (ros::ok()) {
        msg.data = "hello " + std::to_string(idx);
        publisher.publish(msg);

        idx++;
        rate.sleep();
    }


    return 0;
}

File: ros/ws02/first_ws/src/topic_pkg/scripts/publisher.py
#! /usr/bin/env python
# coding=utf8

import rospy
from std_msgs.msg import String

if __name__ == '__main__':
    nodeName = "publisher_node"
    rospy.init_node(nodeName)

    topicName = "/zcb01/topic"
    publisher = rospy.Publisher(topicName, String, queue_size=1000)

    msg = String()
    idx = 0
    rate = rospy.Rate(2)
    while not rospy.is_shutdown():
        msg.data = "hello " + str(idx)
        publisher.publish(msg)

        idx += 1
        rate.sleep()
